src/test/java/hudson/plugins/git/browser/AssemblaWebDoCheckURLTest.java

```diff
@@ -43,6 +43,18 @@ public void testInitialChecksOnRepoUrlWithVariable() throws Exception {
         assertThat(assemblaWebDescriptor.doCheckRepoUrl(project, url), is(FormValidation.ok()));
     }
 
+    @Test
+    public void testDomainLevelChecksOnRepoUrlAllowLocalHostnames() throws Exception {
+        String url = "https://localhost/space/git-plugin/git/source";
```

Comment:
I expect that most users running a `localhost` repository server are not using SSL to encrypt the localhost traffic.  If we're acting like this is a valid use case that should not generate a warning message, then let's accept http instead of https.
```suggestion
        String url = "http://localhost/space/git-plugin/git/source";
```

I think this is not a valid use case because Jenkins itself will display a warning to the user if they set the Jenkins URL to be `localhost`.  The user is still allowed to ignore the error text and apply the change.  However, I'm open to allowing this case if the UrlValidator allows it.  As far as I can tell from the automated test, the UrlValidator does not allow it.